src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -402,5 +404,44 @@ protected virtual IEnumerable<SyntaxNode> ExtractNodesIfInHeader(SyntaxNode root
                 yield return foreachStatement;
             }
         }
+
+        public virtual async Task<TSyntaxNode> TryGetDeepInNodeAsync<TSyntaxNode>(
```

Comment:
not a fan of exposing this publicly.  IS there some service that actually needs to call this?

Reply:
Yep, all refactorings that operate on Expressions where we want to allow them no matter how deep they are (add await, name argument, …).

We could always call it with the normal Extract method if the wanted node is expression. But that is problematic for a number of reason. 
- I actually don't know how to make conditions on generic argument without having to have overloads for each expression / arcane refactoring magic (I get .GetType and have a method that says what type Info instances correspond to expressions).
- I'm not sure I want to make the universal call that we want to do this for _all expression based_ refactorings.